bmi_lstm.py

```diff
@@ -300,148 +308,169 @@ def get_output_var_names(self):
+
+    #------------------------------------------------------------ 
+    def get_input_item_count(self):
+        """Get names of input variables."""
+        return len(self._input_var_names)
+
+    #------------------------------------------------------------ 
+    def get_output_item_count(self):
+        """Get names of output variables."""
+        return len(self._output_var_names)
+
+    #-------------------------------------------------------------------
     #-------------------------------------------------------------------
     # BMI: Variable Information Functions
     #-------------------------------------------------------------------
-    #def get_value(self, var_name, dest):
-    def get_value(self, var_name):
-        """Copy of values.
+    #-------------------------------------------------------------------
+    def get_var_name(self, long_var_name):
+                              
+        return self._var_name_map_long_first[ long_var_name ]
+
+    #-------------------------------------------------------------------
+    def get_var_units(self, long_var_name):
+
+        return self._var_units_map[ long_var_name ]
+                                                             
+    #-------------------------------------------------------------------
+    def get_var_type(self, long_var_name):
+        """Data type of variable.
+
         Parameters
         ----------
         var_name : str
             Name of variable as CSDMS Standard Name.
-        dest : ndarray
-            A numpy array into which to place the values.
+
         Returns
         -------
-        array_like
-            Copy of values.
+        str
+            Data type.
         """
-        #dest[:] = self.get_value_ptr(var_name).flatten()
-        return self.get_value_ptr(var_name)
-        #return dest
+        # JG Edit
+        return str(self.get_value_ptr(long_var_name).dtype)
 
-    #-------------------------------------------------------------------
-    def get_value_ptr(self, var_name):
-        """Reference to values.
+    #------------------------------------------------------------ 
+    def get_var_grid(self, name):
+        
+        # JG Edit
+        # all vars have grid 0 but check if its in names list first
+        if name in (self._output_var_names + self._input_var_names):
+            return self._var_grid_id  
+
+    #------------------------------------------------------------ 
+    def get_var_itemsize(self, name):
+        return np.dtype(self.get_var_type(name)).itemsize
+
+    #------------------------------------------------------------ 
+    def get_var_location(self, name):
+        
+        # JG Edit
+        # all vars have location node but check if its in names list first
+        if name in (self._output_var_names + self._input_var_names):
+            return self._var_loc
+
+    #------------------------------------------------------------ 
+    def get_var_nbytes(self, var_name):
```

Comment:
I modified this function to use getsizeof from the system library. I didn't find that .nbytes was working.

        # JMFrame NOTE: Had to import sys for this function
        return sys.getsizeof(self.get_value_ptr(var_name))